connector_elasticsearch/components/adapter.py

```diff
@@ -33,20 +34,35 @@ def _index_name(self):
 
     @property
     def _es_connection_class(self):
-        return elasticsearch.RequestsHttpConnection
+        return elasticsearch.ReqquestsHttpConnection
 
     def _get_es_client(self):
         backend = self.backend_record
-        api_key = (
-            (backend.api_key_id, backend.api_key)
-            if backend.api_key_id and backend.api_key
-            else None
-        )
-        return elasticsearch.Elasticsearch(
-            [backend.es_server_host],
-            connection_class=self._es_connection_class,
-            api_key=api_key,
-        )
+
+        if backend.is_http_authentification:
+            if backend.es_user or backend.es_password:
+                auth = (backend.es_user, backend.es_password)
```

Comment:
If one of them is not filled, it will be `False`. It probably expects an empty string ?

Reply:
i've putted an AND in place of or